src/notifications/reducer.ts

```diff
@@ -0,0 +1,30 @@
+import { createReducer } from 'redux-act'
+
+import * as actions from './actions'
+
+export interface State {
+    readNotificationList: any[]
```

Comment:
I don't see the point in writing the reducer as a TS module but then just using `any` for the state typings. If you type these properly, all the selectors will also get typed.

I am also concerned about the complexity of these states. 
There shouldn't be much need to have two separate `readNotificationList` and `unreadNotificationList` states; just storing a single list of notifications and having `isRead`-like flags on the objects would suffice and allow you to derive these separate lists at selector level.

Also if I understand the `showMoreIndex` state correctly, it means that when you click the "Show more" it will be set to a given index in one of the lists to be able to show all the message for that specific notif. Again I would recommend putting this as a flag on each notif object's state. That would allow more than 1 notif to have "show more" state set, and no need to mess around with either `undefined` or a `number` state value

`unreadNotifications` state should replace the state you added in the overview's reducer (currently this is being unused here, but rename it something like `unreadNotifsCount` and use it instead of the one you're currently using in overview state)